iphone/cli/hooks/storyboard.js

```diff
@@ -0,0 +1,591 @@
+const path = require('path');
+
+exports.cliVersion = '>=3.2.1';
+
+exports.init = function (logger, config, cli) {
+	let storyboardManager = new StoryboardManager(cli, config, logger);
+	storyboardManager.initialize();
+};
+
+/**
+ * Manages all available storyboards from modules.
+ *
+ * Scans for available storyboards and inspects them to get the required info to
+ * properly integrate them into the Xcode project.
+ */
+class StoryboardManager {
+
+	/**
+	 * Constructs a new storyboard manager.
+	 *
+	 * @param {Object} cli - CLI instance
+	 * @param {Object} config - Project configuration
+	 * @param {Object} logger - Logger instance
+	 * @access public
+	 */
+	constructor(cli, config, logger) {
+		this._cli = cli;
+		this._config = config;
+		this._logger = logger;
+		this._builder = null;
+		this._storyboards = new Map();
+	}
+
+	/**
+	 * Initializes the storyboard manager by hooking into the required build steps.
+	 *
+	 * @access public
+	 */
+	initialize() {
+		this._cli.on('build.pre.compile', {
+			priority: 1200,
+			post: (builder, callback) => {
+				this._logger.trace('Starting storyboard detection');
+				this._builder = builder;
+				this.detectStoryboards().then(callback, e => {
+					if (this._cli.argv.platform === 'ios') {
+						this._logger.error(e.stack);
+					}
+					callback(e);
+				});
+			}
+		});
+
+		this._cli.on('build.ios.xcodeproject', {
+			pre: this.addStoryboardsToXcodeProject.bind(this)
+		});
+	}
+
+	/**
+	 * Detects all available storyboards.
+	 *
+	 * @return {Promise<void>}
+	 * @access private
+	 */
+	async detectStoryboards() {
+		const storyboardPaths = await this.findStoryboardPaths();
+		if (storyboardPaths.length === 0) {
+			return;
+		}
+
+		const incrementalDirectory = path.join(this._builder.projectDir, 'build', 'incrementalStoryboards');
+		const outputDirectory = path.join(this._builder.projectDir, 'build', 'inspectStoryboards');
+		const task = new InspectStoryboardsTask({
+			name: 'ti:inspectStoryboards',
+			logger: this._logger,
+			incrementalDirectory
+		});
+		task.outputDirectory = outputDirectory;
+		storyboardPaths.forEach(storyboardPath => {
+			task.addStoryboardPath(storyboardPath);
+		});
+		task.postTaskRun = () => {
+			this._storyboards = task.storyboards;
+
+			// Convert the internal ES6 map to an object to avoid ES6 in the builder
+			const storyboardObject = {};
+			this._storyboards.forEach(storyboardInfo => {
+				storyboardObject[storyboardInfo.name] = {
+					name: storyboardInfo.name,
+					path: storyboardInfo.path
+				};
+			});
+			this._builder.storyboards = storyboardObject;
+		};
+		return task.run();
+	}
+
+	/**
+	 * Finds any .storyboard file inside modules.
+	 *
+	 * @return {Promise<string[]>} Promise resolving to array of available storyboard paths
+	 * @access private
+	 */
+	async findStoryboardPaths() {
+		const pathsToScan = [];
+		for (let module of this._builder.modules) {
+			pathsToScan.push(path.join(module.modulePath, 'platform'));
```

Comment:
Do you also want to check `platform/ios`, `platform/iphone`, `Resources/ios`, `Resources/iphone`? Because `this.scanPathForStoryboards()` isn't recursive, so you'd have to add those subdirs here to the array to check.

Reply:
The storyboard is only getting fetched from modules. In modules there should not be platform/ios, platform/iphone, Resources/ios, Resources/iPhone as modules are platform specific.